istioctl/cmd/istiod-list.go

```diff
@@ -0,0 +1,312 @@
+			l := clog.NewConsoleLogger(cmd.OutOrStdout(), cmd.ErrOrStderr(), scope)
+			return istiodList(cmd.OutOrStdout(), listArgs, kubeConfigFlags, l)
+		}}
+
+	listCmd.PersistentFlags().StringVarP(&listArgs.manifestsPath, "manifests", "d", "", mesh.ManifestsFlagHelpStr)
+	return listCmd
+}
+
+func istiodList(writer io.Writer, listArgs *istiodListArgs, restClientGetter genericclioptions.RESTClientGetter, l clog.Logger) error {
+	restConfig, err := restClientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	iops, err := getIOPs(restConfig)
+	if err != nil {
+		return err
+	}
+	return printIOPs(writer, iops, listArgs.manifestsPath, restConfig, l)
+}
+
+func printIOPs(writer io.Writer, iops []*iopv1alpha1.IstioOperator, manifestsPath string, restConfig *rest.Config, l clog.Logger) error {
+	if len(iops) == 0 {
+		_, err := fmt.Fprintf(writer, "No IstioOperators present.\n")
+		return err
+	}
+
+	sort.Slice(iops, func(i, j int) bool {
+		return iops[i].Spec.Revision < iops[j].Spec.Revision
+	})
+
+	w := new(tabwriter.Writer).Init(writer, 0, 8, 1, ' ', 0)
+	fmt.Fprintln(w, "VERSION\tREVISION\tPROFILE\tCUSTOMIZATIONS\tPODS\tAGE")
+	for _, iop := range iops {
+		podCount, deploymentAge, err := getControlPlaneDeployment(iop, manifestsPath, restConfig)
+		if err != nil {
+			podCount = "<error>"
+			deploymentAge = "<error>"
+		}
+
+		diffs, err := getDiffs(iop, manifestsPath, effectiveProfile(iop.Spec.Profile), l)
+		if err != nil {
+			return err
+		}
+		for i, diff := range diffs {
+			if i == 0 {
+				fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\n",
+					iop.Spec.Tag,
+					renderWithDefault(iop.Spec.Revision, "master"),
+					renderWithDefault(iop.Spec.Profile, "default"),
+					diff, podCount, deploymentAge)
+			} else {
+				fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\n",
+					"",
+					"",
+					"",
+					diff, "", "")
+			}
+		}
+	}
+	return w.Flush()
+}
+
+func getIOPs(restConfig *rest.Config) ([]*iopv1alpha1.IstioOperator, error) {
+	client, err := dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return nil, err
+	}
+	ul, err := client.
+		Resource(istioOperatorGVR).
+		Namespace(istioNamespace).
+		List(context.TODO(), meta_v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	iops := []*iopv1alpha1.IstioOperator{}
+	for _, un := range ul.Items {
+		un.SetCreationTimestamp(meta_v1.Time{}) // UnmarshalIstioOperator chokes on these
+		by := util.ToYAML(un.Object)
+		iop, err := operator_istio.UnmarshalIstioOperator(by, true)
+		if err != nil {
+			return nil, err
+		}
+		iops = append(iops, iop)
+	}
+	return iops, nil
+}
+
+func getDiffs(installed *iopv1alpha1.IstioOperator, manifestsPath, profile string, l clog.Logger) ([]string, error) {
+	setFlags := []string{"profile=" + profile}
+	if manifestsPath != "" {
+		setFlags = append(setFlags, fmt.Sprintf("installPackagePath=%s", manifestsPath))
+	}
+
+	_, base, err := manifest.GenerateConfig([]string{}, setFlags, true, nil, l)
+	if err != nil {
+		return []string{}, err
```

Comment:
This should be a `[]string`. or may declare it once and use it across the func.

```go
var emptyDiff []string
...
if err != nil {
   return emptyDiff, err
}
```

Reply:
Is this part of a style guide?  I find it clear to see that nothing is being returned using a literal.  The compiler will take care of optimizing.  Why define a pseudo-constant for error returns?